langchain/vectorstores/filters/base.py

```diff
@@ -0,0 +1,17 @@
+"""Interface for vector store Filters."""
+from __future__ import annotations
+
+from abc import ABC, abstractmethod
+from typing import Any, Iterable, List, Optional
+
+
+class VectorStoreFilter(ABC):
```

Comment:
how much of this is actually generic vs just being needed for elastic search?

Reply:
As for this, This is basically a FilterBuilder, to build filters for a specific datastore. As I went through all the vectorStores, they actually support filtering before ANN search (except FAISS, not sure)
The idea of this interface is only to build a filter, for different types of vectorstores and then build the final query